frontend/src/entities/dataset/dataset.service.ts

```diff
@@ -185,28 +185,37 @@ export async function uploadFiles (
 }
 
 export async function createDataset (dataset: IDataset) {
-    const requestUrl = '/dataset/create';
-    const payload = {
-        datasetName: dataset.name,
-        datasetType: dataset.type,
-        datasetScope: dataset.scope,
-        datasetDescription: dataset.description,
-        datasetFormat: dataset.format,
-    };
-    const datasetType = dataset.type;
-    const datasetScope = dataset.scope;
     // Scope and Type should always be set for this Dataset object as part of form validation
-    if (datasetType && datasetScope){
+    if (dataset.type && dataset.scope){
+        const requestUrl = '/dataset/create';
+        const payload = {
+            datasetName: dataset.name,
+            datasetType: dataset.type,
+            datasetScope: dataset.scope,
+            datasetDescription: dataset.description,
+            datasetFormat: dataset.format,
+        };
         const headerConfig = {
             headers: {
-                'x-mlspace-dataset-type': datasetType,
-                'x-mlspace-dataset-scope': datasetScope,
+                'x-mlspace-dataset-type': dataset.type,
+                'x-mlspace-dataset-scope': dataset.scope,
             },
         };
         return axios.post(requestUrl, payload, headerConfig);
     }
 }
 
+export const createDatasetHandleAlreadyExists = (dataset: IDataset) => {
+    let unexpectedError = false;
+    createDataset(dataset).catch((error) => {
+        const expectedError = `Bad Request: Dataset ${dataset.name} already exists.`;
+        // Any error that the dataset already existing is unexpected
+        unexpectedError = expectedError !== error.response.data;
+    });
+    
+    return unexpectedError;
+};
+
```

Comment:
Why not just swallow expected errors and throw the unexpected ones? Is there a reason we're returning it and relying on the caller to know that/do something with it?

Reply:
You're thinking I do the expected error check and if it's an unexpected error I just re-raise it, and then have the call to `createDatasetHandleAlreadyExists` be invoked in a try/catch right?